### PR TITLE
foundationdb: add branch 6.2

### PR DESCRIPTION
**Pull request by vlche**
###### Motivation for this change
A new branch of Foundationdb exists for quite a long time, current branch is 6.2.25, so this PR adds support for it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by lostnet — COMMENTED** [review]
It all looks functionally good to me, I've made a few comments on places where I think there's a clearer or preferred way to do something (to make it more maintainable or quicker to review.)

**Comment by SuperSandro2000** [comment]
ping @thoughtpolice

**Comment by eyeinsky** [comment]
Is this otherwise still working (are you using it)? @SuperSandro2000 @lostnet

**Comment by lostnet** [comment]
@eyeinsky  It was a starting point for me, but I wanted 6.3+ for arm64 support. I think I was last using [this](https://github.com/lostnet/nixpkgs/commit/ce959545d9fb982107a7a3bb982bc36d5daa9ced) but since then the project roadmap changed and I wont need foundationdb.

File: pkgs/servers/foundationdb/cmake62.nix [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          ];
+
+        inherit patches;
+
+        # fix up the use of the very weird and custom 'fdb_install' command by just
+        # replacing it with cmake's ordinary version.
+        postPatch = ''
+        for x in bindings/c/CMakeLists.txt fdbserver/CMakeLists.txt fdbmonitor/CMakeLists.txt fdbbackup/CMakeLists.txt fdbcli/CMakeLists.txt; do
+            substituteInPlace $x --replace 'fdb_install' 'install'
+        done
```
Comment (SuperSandro2000) [inline pkgs/servers/foundationdb/cmake62.nix:71]:
```suggestion
          for x in bindings/c/CMakeLists.txt fdbserver/CMakeLists.txt fdbmonitor/CMakeLists.txt fdbbackup/CMakeLists.txt fdbcli/CMakeLists.txt; do
            substituteInPlace $x --replace 'fdb_install' 'install'
          done
```

File: pkgs/servers/foundationdb/cmake62.nix [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        separateDebugInfo = true;
+        enableParallelBuilding = true;
+        dontFixCmake = true;
+
+        cmakeFlags =
+          [ "-DCMAKE_BUILD_TYPE=Release"
+            (lib.optionalString officialRelease "-DFDB_RELEASE=TRUE")
+
+            # FIXME: why can't openssl be found automatically?
```
Comment (SuperSandro2000) [inline pkgs/servers/foundationdb/cmake62.nix:46]:
```suggestion
        cmakeFlags = [
          "-DCMAKE_BUILD_TYPE=Release"
          (lib.optionalString officialRelease "-DFDB_RELEASE=TRUE")

          # FIXME: why can't openssl be found automatically?
```
and so on

File: pkgs/servers/foundationdb/cmake62.nix [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+, gccStdenv, llvmPackages
+, useClang ? false
+, ...
+}:
+
+let
+  stdenv = if useClang then llvmPackages.libcxxStdenv else gccStdenv;
+
+  tests = with builtins;
+    builtins.replaceStrings [ "\n" ] [ " " ] (lib.fileContents ./test-list.txt);
```
Comment (SuperSandro2000) [inline pkgs/servers/foundationdb/cmake62.nix:15]:
```suggestion
  tests = builtins.replaceStrings [ "\n" ] [ " " ] (lib.fileContents ./test-list.txt);
```

**Pull request closed without merging**